indexing-service/src/main/java/io/druid/indexing/common/task/IndexTask.java

```diff
@@ -186,7 +193,15 @@ public TaskStatus run(TaskToolbox toolbox) throws Exception
     final GranularitySpec granularitySpec = ingestionSchema.getDataSchema().getGranularitySpec();
     final int targetPartitionSize = ingestionSchema.getTuningConfig().getTargetPartitionSize();
 
-    final TaskLock myLock = Iterables.getOnlyElement(getTaskLocks(toolbox));
+    final TaskLock myLock;
+    // Confirm we have a lock and it has not been revoked by a higher priority task
+    try {
```

Comment:
this code is in an awful lot of places, it could be worth having it in a helper method.